Fw/Types/StringUtils.cpp

```diff
@@ -29,3 +29,30 @@ U32 Fw::StringUtils::string_length(const CHAR* source, U32 max_len) {
     }
     return length;
 }
+
+FwSignedSizeType Fw::StringUtils::substring_find(const CHAR* source_string, FwSizeType source_size, const CHAR* sub_string, FwSizeType sub_size) {
+
+    FW_ASSERT(source_string != nullptr);
+    FW_ASSERT(sub_string != nullptr);
+
+    // zero size sub-strings should always match
+    if ((source_size > 0) && (0 == sub_size)) {
+        return 0;
+    }
+
+    for (FwSizeType source_index = 0; source_index < source_size; source_index++) {
```

Comment:
Can't you get away with looping until `source_size - sub_size`.  It is impossible to match a substring that would go past the end of your source.

Reply:
source_size - sub_size + 1....to catch the equality case